shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -88,100 +186,191 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
-            throw new InvalidMessageException("Sending appends without setting up the append.");
+            throw new InvalidMessageException("Sending appends without setting up the append. Append Writer id: " + append.getWriterId());
         }
         if (append.getEventNumber() <= session.lastEventNumber) {
-            throw new InvalidMessageException("Events written out of order. Received: " + append.getEventNumber()
-            + " following: " + session.lastEventNumber);
+            throw new InvalidMessageException("Events written out of order. Received: " + append.getEventNumber() +
+            " following: " + session.lastEventNumber +
+            " for Writer id: " + append.getWriterId());
         }
         if (append.isConditional()) {
             throw new IllegalArgumentException("Conditional appends should be written via a ConditionalAppend object.");
         }
-        Preconditions.checkState(bytesLeftInBlock == 0 || bytesLeftInBlock > TYPE_PLUS_LENGTH_SIZE,
-                "Bug in CommandEncoder.encode, block is too small.");
     }
 
-    private void breakFromAppend(ByteBuf data, ByteBuf pendingData, ByteBuf out) {
-        if (bytesLeftInBlock != 0) {
-            if (data != null) {
-                writeMessage(new PartialEvent(data), out);
-            } else {
-                writeMessage(new Padding(bytesLeftInBlock - TYPE_PLUS_LENGTH_SIZE), out);
-            }
-            Session session = setupSegments.get(new SimpleImmutableEntry<>(segmentBeingAppendedTo, writerIdPerformingAppends));
-            writeMessage(new AppendBlockEnd(session.id,
-                    currentBlockSize - bytesLeftInBlock,
-                    pendingData,
-                    session.eventCount,
-                    session.lastEventNumber, session.requestId), out);
-            bytesLeftInBlock = 0;
-            currentBlockSize = 0;
-            session.eventCount = 0;
-            tokenCounter.incrementAndGet();
+    /**
+     * Check if the network channel is idle or not.
+     *
+     * @return true if channel is not used by any writers; false otherwise.
+     */
+    private boolean isChannelFree() {
+        return writerIdPerformingAppends == null;
+    }
+
+    /**
+     * Check/Confirm the Network channel Ownership.
+     * The Network channel ownership is associated with Writer ID and segment to which writes are performed.
+     *
+     * @param writerID  Writers UUID.
+     * @param segment   Segment.
+     * @return true if channel is used by the given WriterId and Segment; false otherwise.
+     */
+    private boolean isChannelOwner(UUID writerID, String segment) {
+        return writerID.equals(writerIdPerformingAppends) && segment.equals(segmentBeingAppendedTo);
+    }
+
+    /**
+     * Start the Append.
+     * if the Append batch size tracker decides the block size
+     * which is more than event message size then timeout will be initiated.
```

Comment:
How is a timeout initiated and how is it handled? Does it block the current call? Does it throw an exception? The code seems to indicate it will schedule a task with a delayed start time. 

Let's be specific about the behavior in the documentation.

Reply:
Its uses the existing timer behavior; The append starts with timer, before the timeout if the append completes then no need to flush any data; otherwise, all pending session data is flushed. I have updated the documentation in blocktimeout run method and other places too.